lib/upload_to_s3.js

```diff
@@ -0,0 +1,48 @@
+import https from 'https';
+import url from 'url';
+
+export default async function uploadToS3 (task, sourceStream, artifactName, expires, httpsHeaders) {
+  var queue = task.runtime.queue;
+
+  // Create date when this artifact should expire (see config).
+  var expiration =
```

Comment:
Could it be possible to just pass in a Date object for expires and let the caller calculate what the expiration should be for the artifact instead of passing in a millisecond value?

You could then assert that expires is an instance of Date before continuing.